modules/common/memcached/memcached.go

```diff
@@ -0,0 +1,52 @@
+package memcached
+
+import (
+	"context"
+	"fmt"
+
+	memcachedv1 "github.com/openstack-k8s-operators/infra-operator/apis/memcached/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewMemcached function will return a memcached v1beta1 instance of the CR. This can then be used
+// with controllerutil.CreateOrPatch() to create the resource in the cluster.
+func NewMemcached(name string, namespace string) *memcachedv1.Memcached {
+	return &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind:       "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
+
+// GetMemcachedPodList gets a list of pods matching the labels provided and returns a slice of pod names.
+func GetMemcachedPodList(ctx context.Context, h *helper.Helper, memcachedName string, namespace string) ([]string, error) {
```

Comment:
If we name this PodList then IMO this should return PodList instead of list of strings.
If we return these strings then we probably have to use a different naming which better represents what these strings are

Reply:
Fair point. I'll change the name